decidim-proposals/lib/decidim/proposals/markdown2proposals.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+require "redcarpet"
+
+module Decidim
+  module Proposals
+    # This class parses a participatory text document in markdown and
+    # produces Proposalsin the form of sections and articles
+    class Markdown2Proposals < ::Redcarpet::Render::Base
+      # Public: Initializes the serializer with a proposal.
+      def initialize(component)
+        super()
+        @component = component
+        @last_position = 0
+        @num_sections = 0
+      end
+
+      def parse(document)
+        renderer = self
+        parser = ::Redcarpet::Markdown.new(renderer)
+        parser.render(document)
+      end
+
+      ##########################################
+      # Redcarpet callbacks
+      ##########################################
+
+      def preprocess(full_document)
+        full_document
+      end
+
+      def postprocess(full_document)
+        full_document
+      end
+
+      def doc_header; end
+
+      def doc_footer; end
+
+      # def normal_text(text)
+      #   puts "normal_text: #{text}"
+      #    text
+      # end
+
+      # def block_code(code, language)
+      #   puts "block_code #{code}[#{language}]"
+      #    "block_code #{code}[#{language}]"
+      # end
+
+      # def codespan(code)
+      #   puts "codespan #{code}"
+      #    "codespan #{code}"
+      # end
+
+      def header(title, level)
+        pt_level = if level > 1
```

Comment:
`s/pt_level/participatory_text_level`

Reply:
@tramuntanal please, avoid these reactions:

![](https://i.imgur.com/FSWXyIC.png)

Reviewing this "stupid" things, which we won't accept under any circumstance, takes a lot of time from our part and it's a simple find & replace from your part. Please understand `pt_level` is in no way clear of what it does, and takes a toll when maintaining the code. 

Thanks.